### PR TITLE
Add must-gather collection script for DSP

### PR DESCRIPTION
**Pull request by DharmitD**
## Description

Adding a must-gather collection script for Data Science Pipelines Application instances.

## How Has This Been Tested?

- Retrieved this image from the CI build for this PR: `quay.io/opendatahub/must-gather:pr-798`
- Used this image to run must-gather on a cluster with DSP Operator and Application installed:
`oc adm must-gather --image=quay.io/opendatahub/must-gather:pr-798`
## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: must-gather/collection-scripts/gather-data-science-pipelines [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+# Get logs from all Data Science Pipelines Application namespaces
+for i in $datasciencepipelines;
+do
+# Get pod logs for all the pods in Data Science Pipelines Application namespaces
+oc adm inspect namespace/"$i" --dest-dir=must-gather || echo "Error getting logs from $i"
+
+done
+
+## Get Operator pod logs
```
Comment (gmfrasca) [inline must-gather/collection-scripts/gather-data-science-pipelines:13]: shouldn't this be for the DSP Operator, not ODH operator?

Comment (DharmitD) [reply must-gather/collection-scripts/gather-data-science-pipelines]: Done, updated to have this for DSP Operator.

Comment (DharmitD) [reply must-gather/collection-scripts/gather-data-science-pipelines]: Updating to take must gather for DSP Operator out, per discussion with @VaishnaviHire . If any application is deployed using kfdef we already collect the pod logs, hence this is not required.

**Comment by VaishnaviHire** [comment]
@DharmitD PR needs a rebase

**Review by VaishnaviHire — APPROVED** [review]
/lgtm

**Pull request merged**